ceph/rados/core_workflows.py

```diff
@@ -1498,6 +1498,31 @@ def fetch_host_node(self, daemon_type: str, daemon_id: str = None) -> object:
             )
             return None
 
+    def fetch_host_config(self, daemon_type: str, daemon_id: str = None) -> object:
+        """
+        Fetches the config of the given daemon in a cluster
+
+        Args:
+            daemon_type (str): type of daemon
+            daemon_id (str, optional): name of the daemon, ID in case of OSD's
+
+        Returns:
+            dict: host config
+        """
+        json_str = None
+        if daemon_type == "osd":
```

Comment:
This method will only cater for OSDs. We need a general method that would work with any daemons passed.

Please re-write the method so that it works for any daemons